site/app/libraries/database/DatabaseQueries.php

```diff
@@ -5403,6 +5403,84 @@ public function getUserNotifications($user_id, $show_all) {
         return $results;
     }
 
+    /**
+     * Get 10 most recent notifications in each unarchived course
+     * @param string $user_id
+     * @param array $courses
+     */
+    public function getAllRecentUserNotifications($user_id, $courses) {
+        $all_rows = [];
+        $original_config = clone $this->core->getConfig();
+        foreach ($courses as $course) {
+            $semester = $course->getTerm();
+            $course_name = $course->getTitle();
+            $this->core->loadCourseConfig($semester, $course_name);
+            $this->core->loadCourseDatabase();
```

Comment:
Instead of doing all of this business logic within `DatabaseQueries.php`, let's move this to a controller instead.  `DatabaseQueries.php` is supposed to be about actually executing queries, not manipulating the core...